presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java

```diff
@@ -535,4 +408,128 @@ public static boolean hasType(ObjectInspector objectInspector, PrimitiveCategory
+        private List<TestColumn> readColumns;
+        private ConnectorSession session = SESSION;
+        private int rowsCount = NUM_ROWS;
+
+        private FileFormatAssertion(String formatName)
+        {
+            this.formatName = requireNonNull(formatName, "formatName is null");
+        }
+
+        public FileFormatAssertion withOutputFormat(HiveOutputFormat<?, ?> outputFormat)
+        {
+            this.outputFormat = requireNonNull(outputFormat, "outputFormat is null");
+            return this;
+        }
+
+        public FileFormatAssertion withInputFormat(InputFormat<?, ?> inputFormat)
+        {
+            this.inputFormat = requireNonNull(inputFormat, "inputFormat is null");
+            return this;
+        }
+
+        public FileFormatAssertion withSerde(SerDe serde)
+        {
+            this.serde = requireNonNull(serde, "serde is null");
+            return this;
+        }
+
+        public FileFormatAssertion withCompressionCodec(String compressionCodec)
+        {
+            this.compressionCodec = Optional.of(requireNonNull(compressionCodec, "compressionCodec is null"));
+            return this;
+        }
+
+        public FileFormatAssertion withColumns(List<TestColumn> inputColumns)
+        {
+            withWriteColumns(inputColumns);
+            withReadColumns(inputColumns);
+            return this;
+        }
+
+        public FileFormatAssertion withWriteColumns(List<TestColumn> writeColumns)
+        {
+            this.writeColumns = requireNonNull(writeColumns, "writeColumns is null");
+            return this;
+        }
+
+        public FileFormatAssertion withReadColumns(List<TestColumn> readColumns)
+        {
+            this.readColumns = requireNonNull(readColumns, "readColumns is null");
+            return this;
+        }
+
+        public FileFormatAssertion withRowsCount(int rowsCount)
+        {
+            this.rowsCount = rowsCount;
+            return this;
+        }
+
+        public FileFormatAssertion withSession(ConnectorSession session)
+        {
+            this.session = requireNonNull(session, "session is null");
+            return this;
+        }
+
+        public FileFormatAssertion isReadableByPageSource(HivePageSourceFactory pageSourceFactory)
+                throws Exception
+        {
+            testReadValues(ImmutableList.of(pageSourceFactory), ImmutableList.of());
+            return this;
+        }
+
+        public FileFormatAssertion isReadableByRecordCursor(HiveRecordCursorProvider cursorProvider)
+                throws Exception
+        {
+            testReadValues(ImmutableList.of(), ImmutableList.of(cursorProvider));
+            return this;
+        }
+
+        private void testReadValues(List<HivePageSourceFactory> pageSourceFactories, List<HiveRecordCursorProvider> cursorProviders)
+                throws Exception
+        {
+            assertNotNull(outputFormat, "outputFormat must be specified");
```

Comment:
I think  you can move this assertions to AssertJ. It was added as presto dependency some time ago.